### PR TITLE
Bindu | BAH-439 | Person attributes with coded concept answers displa…

### PR DESCRIPTION
**Pull request by binduak**
…y concept IDs on registration search
https://bahmni.atlassian.net/browse/BAH-439

### CODE REVIEW

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java [hunk 1]
```diff
@@ -80,7 +84,18 @@ private void mapPersonAttributes(Patient patient, List<String> patientSearchResu
         String queriedPersonAttributes = patientSearchResultFields.stream()
                 .map(attributeName -> {
                     PersonAttribute attribute = patient.getAttribute(attributeName);
-                    return attribute == null ? null : formKeyPair(attributeName, attribute.getValue());
+                    if(attribute != null) {
+                        if("org.openmrs.Concept".equals(attribute.getAttributeType().getFormat())) {
+                            Concept concept = Context.getConceptService().getConcept(attribute.getValue());
+                            ConceptName shortNameInLocale = concept.getShortNameInLocale(Context.getLocale());
```
[the remark this answers is not in the document]
Comment (binduak) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java]: @angshu I don't see much difference. 'Context.getLocale()' intern calls the 'Context.getUserContext().getLocale()'
https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/context/Context.java#L727

[the remark this answers is not in the document]
Comment (binduak) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java]: Changed the shortName with FSN and updated the PR.

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java [hunk 2]
```diff
@@ -80,7 +84,18 @@ private void mapPersonAttributes(Patient patient, List<String> patientSearchResu
         String queriedPersonAttributes = patientSearchResultFields.stream()
                 .map(attributeName -> {
                     PersonAttribute attribute = patient.getAttribute(attributeName);
-                    return attribute == null ? null : formKeyPair(attributeName, attribute.getValue());
+                    if(attribute != null) {
+                        if("org.openmrs.Concept".equals(attribute.getAttributeType().getFormat())) {
+                            Concept concept = Context.getConceptService().getConcept(attribute.getValue());
+                            ConceptName fullySpecifiedName = concept.getFullySpecifiedName(Context.getLocale());
```
Comment (binduak) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java]: @angshu I have changed shortName with FSN and updated the PR.

Comment (binduak) [inline bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java]: I don't see much difference when using 'Context.getLocale()' and 'Context.getUserContext().getLocale()'. 'Context.getLocale()' internally calls 'getUserContext().getLocale()' method. 
https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/context/Context.java#L727

File: bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (angshu) [reply bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/mapper/PatientResponseMapper.java]: Seems context.getLocale() uses 'Context.getUserContext'. Are you saying that login in as a user with a different locale (different from that set in system), has got no effect? Also another question is = 'what do we expect if the concept does not have a FSN defined in the the user's language in concept dictionary?'

**Pull request merged**